aws/resource_aws_securityhub_standards_control_test.go

```diff
@@ -0,0 +1,157 @@
+
+	resourceName := "aws_securityhub_standards_control.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSecurityHubStandardsControlExists(resourceName, standardsControl),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccAWSSecurityHubStandardsControlConfig_enabledControlStatus(),
+				ExpectError: regexp.MustCompile("InvalidInputException: DisabledReason should not be given for action other than disabling control"),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSSecurityHubStandardsControlExists(n string, control *securityhub.StandardsControl) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).securityhubconn
+
+		arn := rs.Primary.ID
+		subscription_arn := path.Dir(strings.ReplaceAll(arn, "control", "subscription"))
+
+		resp, err := conn.DescribeStandardsControls(&securityhub.DescribeStandardsControlsInput{
+			StandardsSubscriptionArn: aws.String(subscription_arn),
+		})
+		if err != nil {
+			return fmt.Errorf("error reading Security Hub %s standard controls: %s", subscription_arn, err)
+		}
+
+		controlNotFound := true
+
+		for _, c := range resp.Controls {
+			if aws.StringValue(c.StandardsControlArn) != arn {
+				continue
+			}
+
+			controlNotFound = false
+			control = c
+		}
+
+		if controlNotFound {
+			return fmt.Errorf("Security Hub %s standard control %s not found", subscription_arn, arn)
+		}
+
+		return nil
+	}
+}
+
+func testAccAWSSecurityHubStandardsControlConfig_basic() string {
+	return composeConfig(
+		testAccAWSSecurityHubStandardsSubscriptionConfig_basic,
+		`
+resource aws_securityhub_standards_control test {
+  standards_control_arn = "arn:aws:securityhub:us-east-1:078884324539:control/cis-aws-foundations-benchmark/v/1.2.0/1.10"
+  control_status        = "ENABLED"
+  depends_on            = [aws_securityhub_standards_subscription.test]
+}
+`)
+}
+
+func testAccAWSSecurityHubStandardsControlConfig_disabledControlStatus() string {
+	return composeConfig(
+		testAccAWSSecurityHubStandardsSubscriptionConfig_basic,
+		`
+resource aws_securityhub_standards_control test {
+  standards_control_arn = "arn:aws:securityhub:us-east-1:078884324539:control/cis-aws-foundations-benchmark/v/1.2.0/1.11"
+  control_status        = "DISABLED"
+  disabled_reason       = "We handle password policies within Okta"
+  depends_on            = [aws_securityhub_standards_subscription.test]
+}
+`)
+}
+
+func testAccAWSSecurityHubStandardsControlConfig_enabledControlStatus() string {
+	return composeConfig(
+		testAccAWSSecurityHubStandardsSubscriptionConfig_basic,
+		`
+resource aws_securityhub_standards_control test {
+  standards_control_arn = "arn:aws:securityhub:us-east-1:078884324539:control/cis-aws-foundations-benchmark/v/1.2.0/1.12"
```

Comment:
Avoid hard-coding account ID.
We can also then remove the explicit `depends_on`.

```suggestion
  standards_control_arn = format("%s/1.12", replace(aws_securityhub_standards_subscription.test.id, "subscription", "control"))
```